lib/partitioninfo.coffee

```diff
@@ -54,22 +69,34 @@ getLogicalPartitions = (disk, index, offset, extendedPartitionOffset = offset, l
 getPartitions = (disk, options) ->
 	options = _.defaults(options, getLogical: true)
 	disk = Promise.promisifyAll(disk)
-	result = []
+	result = {}
 	extended = null
-	readMbr(disk, options.offset)
-	.then (buf) ->
-		for p, index in getPartitionsFromMBRBuf(buf)
-			if MBR.Partition.isExtended(p.type)
-				extended = p
-				if options.includeExtended
-					result.push(partitionDict(p, options.offset, index + 1))
-			else
-				result.push(partitionDict(p, options.offset, index + 1))
-		if extended != null and options.getLogical
-			return getLogicalPartitions(disk, MBR_FIRST_LOGICAL_PARTITION, extended.byteOffset())
-			.then (logicalPartitions) ->
-				result.push(logicalPartitions...)
+	readFromDisk(disk, options.offset, MBR_SIZE)
+	.then (mbrBuf) ->
+		partitions = getPartitionsFromMBRBuf(mbrBuf)
+		if partitions.length == 1 and partitions[0].type == GPT_PROTECTIVE_MBR
```

Comment:
Super NIT: Can we have these as `is`? I find it a lot more readable, and we use it elsewhere.

Reply:
udpated